llvm/test/Analysis/LoopAccessAnalysis/depend_diff_types.ll

```diff
@@ -194,3 +194,84 @@ loop:
 exit:
   ret void
 }
+
+; In the following test, the source is loop-invariant.
+
+define void @type_size_equivalence_src_loopinv(ptr nocapture %vec, i64 %n) {
+; CHECK-LABEL: 'type_size_equivalence_src_loopinv'
+; CHECK-NEXT:    loop:
+; CHECK-NEXT:      Memory dependences are safe
+; CHECK-NEXT:      Dependences:
+; CHECK-NEXT:      Run-time memory checks:
+; CHECK-NEXT:      Grouped accesses:
+; CHECK-EMPTY:
+; CHECK-NEXT:      Non vectorizable stores to invariant address were not found in loop.
+; CHECK-NEXT:      SCEV assumptions:
+; CHECK-EMPTY:
+; CHECK-NEXT:      Expressions re-written:
+;
+entry:
+  br label %loop
+
+loop:
+  %iv = phi i64 [ 0, %entry ], [ %iv.next, %loop ]
+
+  %gep.iv = getelementptr i64, ptr %vec, i64 %iv
+  %ld.i64 = load i64, ptr %gep.iv, align 8
+
+  %gep.n = getelementptr inbounds i64, ptr %vec, i64 %n
```

Comment:
Moving the `GEP` here out of the loop would probably make it easier to see that its loop invariant